akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -473,7 +473,9 @@ private[remote] class Association(
           case (((q, m), c), w) ⇒ ((q, m), (c, w))
         }
 
-      val (mergeHub, aeronSinkCompleted) = MergeHub.source[EnvelopeBuffer].toMat(transport.aeronSink(this))(Keep.both).run()(materializer)
+      val (mergeHub, aeronSinkCompleted) = MergeHub.source[EnvelopeBuffer]
+        .via(transport.killSwitch.flow[EnvelopeBuffer])
```

Comment:
this is not enough, I also need to make sure that all parts are teared down if one of them fails